consoles/sshXtermIPMI.pm

```diff
@@ -55,4 +60,19 @@ sub do_mc_reset ($self) {
     return;
 }
 
+sub current_screen ($self) {
+    my $ret;
+    my $retry = 0;
+    my $max_errs = $bmwqemu::vars{IPMI_SOL_MAX_RECONNECTS} // 5;
+
+    while (1) {
+        $ret = $self->SUPER::current_screen;
+        return $ret if waitpid($self->{xterm_pid}, WNOHANG) == 0;
+        die 'Too many IPMI SOL errors' if ++$self->{reconnects} > $max_errs;
+        bmwqemu::fctwarn("IPMI SOL connection died, reconnect $self->{reconnects} / $max_errs");
+        select(undef, undef, undef, 0.1 * $retry++);    # sleep between retries
```

Comment:
Using `Time::HiRes` would probably be clearer (and we use `Time::HiRes` already anyway).

Reply:
Fixed.